diff -u deltarpm-3.5/makedeltarpm.c deltarpm-pld/makedeltarpm.c
--- deltarpm-3.5/makedeltarpm.c	2009-08-18 15:18:13.000000000 +0200
+++ deltarpm-pld/makedeltarpm.c	2010-06-17 01:01:12.000000000 +0200
@@ -926,6 +929,13 @@
 		    fprintf(vfp, "skipping %s: bad file flags\n", np);
 		  skipped_fileflags++;
 		}
+	      else if ((fileflags[i] & FILE_DOC) != 0)
+		{
+		  if (verbose > 1)
+		    fprintf(vfp, "USING DOC %s\n", np);
+		  lsize = size;
+		  skip = 0;
+		}
 	      else if ((fileverify[i] & (VERIFY_MD5|VERIFY_FILESIZE)) != (VERIFY_MD5|VERIFY_FILESIZE))
 		{
 		  if (verbose > 1)
diff -u deltarpm-3.5/rpmhead.h deltarpm-pld/rpmhead.h
--- deltarpm-3.5/rpmhead.h	2009-07-21 15:41:08.000000000 +0200
+++ deltarpm-pld/rpmhead.h	2009-10-26 00:57:40.000000000 +0100
@@ -38,6 +38,7 @@
 #define SIGTAG_SHA1     269
 
 #define FILE_CONFIG     (1 << 0)
+#define FILE_DOC        (1 << 1)
 #define FILE_MISSINGOK  (1 << 3)
 #define FILE_GHOST      (1 << 6)
 #define FILE_UNPATCHED  (1 << 10)
