--- deltarpm-3.6.3/cfile.c.orig	2020-08-27 17:34:37.000000000 +0200
+++ deltarpm-3.6.3/cfile.c	2023-10-08 09:19:04.378740047 +0200
@@ -654,7 +654,8 @@ cropen_lz(struct cfile *f)
 {
   lzma_stream tmp = LZMA_STREAM_INIT;
   f->strm.lz = tmp;
-  if (lzma_auto_decoder(&f->strm.lz, 1 << 25, 0) != LZMA_OK)
+  //if (lzma_auto_decoder(&f->strm.lz, 1 << 25, 0) != LZMA_OK)
+  if (lzma_auto_decoder(&f->strm.lz, 1 << 27, 0) != LZMA_OK)
     {
       free(f);
       return 0;
@@ -1320,6 +1321,10 @@ cfile_open(int mode, int fd, void *fp, i
 	    comp = CFILE_COMP_LZMA;
 	  else if (f->buf[0] == 0135 && f->buf[1] == 0 && f->buf[2] == 0)
 	    comp = CFILE_COMP_LZMA;
+	  else if (f->buf[0] == 0 && f->buf[1] == 0x5d && f->buf[2] == 0 && f->buf[3] == 0)
+	    comp = CFILE_COMP_LZMA;
+	  else if (f->buf[0] == 0 && f->buf[1] == 0x03 && f->buf[2] == 0 && f->buf[3] == 0)
+	    comp = CFILE_COMP_LZMA;
 	  else if (f->buf[0] == 0xfd && f->buf[1] == '7' && f->buf[2] == 'z' && f->buf[3] == 'X' && f->buf[4] == 'Z')
 	    comp = CFILE_COMP_XZ;
 	  else if ((f->buf[0] & 0xf0) == 0x20 && f->buf[1] == 0xb5 && f->buf[2] == 0x2f && f->buf[3] == 0xfd)
@@ -1463,6 +1468,7 @@ cfile_setlevel(int comp, int level)
     case CFILE_COMP_GZ:
     case CFILE_COMP_GZ_RSYNC:
     case CFILE_COMP_BZ:
+    case CFILE_COMP_LZMA:
       deflevel = 9;
       break;
     default:
--- deltarpm-3.6.3/makedeltarpm.c.orig	2023-10-07 22:04:03.314819811 +0200
+++ deltarpm-3.6.3/makedeltarpm.c	2023-10-08 09:19:37.131895941 +0200
@@ -1084,11 +1084,13 @@ main(int argc, char **argv)
       fprintf(stderr, "payload open failed\n");
       exit(1);
     }
+/*
   if (cfile_detect_rsync(newbz))
     {
       fprintf(stderr, "detect_rsync failed\n");
       exit(1);
     }
+ */
   targetcomp = newbz->comp;
   if ((payloadflags = headstring(d.h, TAG_PAYLOADFLAGS)) != 0)
     {
--- deltarpm-git-20110223/Makefile~	2011-02-19 18:53:23.000000000 +0200
+++ deltarpm-git-20110223/Makefile	2011-11-07 22:29:01.877800526 +0200
@@ -1,7 +1,7 @@
-prefix=/usr/local
+prefix=/usr
 bindir=$(prefix)/bin
 libdir=$(prefix)/lib
-mandir=$(prefix)/man
+mandir=$(prefix)/share/man
 rpmdumpheader=$(bindir)/rpmdumpheader
 zlibdir=zlib-1.2.2.f-rsyncable
 zlibbundled=$(zlibdir)/libz.a
